File: android-sdk-ui/src/main/java/com/braze/ui/inappmessage/factories/DefaultInAppMessageAnimationFactory.java
package com.braze.ui.inappmessage.factories;

import android.content.res.Resources;
import android.view.animation.AlphaAnimation;
import android.view.animation.Animation;

import com.braze.enums.inappmessage.SlideFrom;
import com.braze.models.inappmessage.IInAppMessage;
import com.braze.models.inappmessage.InAppMessageSlideup;
import com.braze.ui.inappmessage.IInAppMessageAnimationFactory;
import com.braze.ui.support.AnimationUtils;

public class DefaultInAppMessageAnimationFactory implements IInAppMessageAnimationFactory {
  private final int mShortAnimationDurationMillis = Resources.getSystem().getInteger(android.R.integer.config_shortAnimTime);

  @Override
  public Animation getOpeningAnimation(IInAppMessage inAppMessage) {
    if (inAppMessage instanceof InAppMessageSlideup) {
      InAppMessageSlideup inAppMessageSlideup = (InAppMessageSlideup) inAppMessage;
      if (inAppMessageSlideup.getSlideFrom() == SlideFrom.TOP) {
        return AnimationUtils.createVerticalAnimation(-1, 0, mShortAnimationDurationMillis, false);
      } else {
        return AnimationUtils.createVerticalAnimation(1, 0, mShortAnimationDurationMillis, false);
      }
    } else {
      return AnimationUtils.setAnimationParams(new AlphaAnimation(0, 1), mShortAnimationDurationMillis, true);
    }
  }

  @Override
  public Animation getClosingAnimation(IInAppMessage inAppMessage) {
    if (inAppMessage instanceof InAppMessageSlideup) {
      InAppMessageSlideup inAppMessageSlideup = (InAppMessageSlideup) inAppMessage;
      if (inAppMessageSlideup.getSlideFrom() == SlideFrom.TOP) {
        return AnimationUtils.createVerticalAnimation(0, -1, mShortAnimationDurationMillis, false);
      } else {
        return AnimationUtils.createVerticalAnimation(0, 1, mShortAnimationDurationMillis, false);
      }
    } else {
      return AnimationUtils.setAnimationParams(new AlphaAnimation(1, 0), mShortAnimationDurationMillis, false);
    }
  }
}

File: android-sdk-ui/src/main/java/com/braze/ui/inappmessage/factories/DefaultInAppMessageAnimationFactory.kt
package com.braze.ui.inappmessage.factories

import android.content.res.Resources
import android.view.animation.AlphaAnimation
import android.view.animation.Animation
import com.braze.enums.inappmessage.SlideFrom
import com.braze.models.inappmessage.IInAppMessage
import com.braze.models.inappmessage.InAppMessageSlideup
import com.braze.ui.inappmessage.IInAppMessageAnimationFactory
import com.braze.ui.support.createVerticalAnimation
import com.braze.ui.support.setAnimationParams

open class DefaultInAppMessageAnimationFactory : IInAppMessageAnimationFactory {
    private val shortAnimationDurationMs =
        Resources.getSystem().getInteger(android.R.integer.config_shortAnimTime).toLong()

    override fun getOpeningAnimation(inAppMessage: IInAppMessage): Animation? {
        return if (inAppMessage is InAppMessageSlideup) {
            if (inAppMessage.slideFrom === SlideFrom.TOP) {
                createVerticalAnimation(-1f, 0f, shortAnimationDurationMs, false)
            } else {
                createVerticalAnimation(1f, 0f, shortAnimationDurationMs, false)
            }
        } else {
            setAnimationParams(AlphaAnimation(0f, 1f), shortAnimationDurationMs, true)
        }
    }

    override fun getClosingAnimation(inAppMessage: IInAppMessage): Animation? {
        return if (inAppMessage is InAppMessageSlideup) {
            if (inAppMessage.slideFrom === SlideFrom.TOP) {
                createVerticalAnimation(0f, -1f, shortAnimationDurationMs, false)
            } else {
                createVerticalAnimation(0f, 1f, shortAnimationDurationMs, false)
            }
        } else {
            setAnimationParams(AlphaAnimation(1f, 0f), shortAnimationDurationMs, false)
        }
    }
}
